src/netbsd.rs

```diff
@@ -31,9 +31,22 @@ pub fn getrandom_inner(dest: &mut [MaybeUninit<u8>]) -> Result<(), Error> {
     // getrandom(2) was introduced in NetBSD 10.0
     static GETRANDOM: Weak = unsafe { Weak::new("getrandom\0") };
     if let Some(fptr) = GETRANDOM.ptr() {
-        let func: GetRandomFn = unsafe { core::mem::transmute(fptr) };
+        let func = fptr.cast::<GetRandomFn>();
+        // SAFETY:
+        // * We assume that any symbol found by `Weak` has
+        //   signature `GetRandomRn`, though that isn't guaranteed.
+        // * TODO: What if `!func.is_aligned()` as required by `as_ref()`?
+        // * TODO: What should we say about it being `dereferenceable`
+        //   besides we know it is non-null?
+        // * We assume that the pointer remains valid for the lifetime of the
+        //   process; i.e. we assume nothing unloads the library providing
+        //   `getrandom` or otherwise causes it to become uncallable.
+        //
+        // Cryptographic security: We assume any implementation of `getrandom`
+        // we find is libc's `getrandom` though this isn't guaranteed.
+        let func: GetRandomFn = unsafe { *func.as_ref() };
```

Comment:
I don't think this code is correct. You transform function pointer to `*mut GetRandomFn`, dereference it, and then use it as a function. AFAIK `transmute` currently is the only way to transform raw pointers to function pointers.

Reply:
Note that my multiple wrong implementations passed CI because we don't actually run the tests on NetBSD, which is the only user of this code right now.